File: pbmx_ffi/java/io/rmf/pbmx/Fingerprint.java
package io.rmf.pbmx;

import io.rmf.pbmx.ffi.RawFingerprint;

public final class Fingerprint {
    RawFingerprint raw;

    Fingerprint(RawFingerprint raw) {
        this.raw = raw;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        for (byte b : this.raw.bytes) {
            sb.append(String.format("%02x", b));
        }
        return sb.toString();
    }
}

File: pbmx_droid/app/src/main/java/io/rmf/pbmx/Fingerprint.kt
package io.rmf.pbmx

import io.rmf.pbmx.ffi.RawFingerprint

class Fingerprint internal constructor(internal var raw: RawFingerprint): Comparable<Fingerprint> {

    override fun equals(other: Any?): Boolean {
        return this.toString() == other.toString()
    }

    override fun hashCode(): Int {
        return this.toString().hashCode()
    }

    override fun compareTo(other: Fingerprint): Int {
        return this.toString().compareTo(other.toString())
    }

    override fun toString(): String {
        return this.raw.bytes
            .fold(StringBuilder()) { sb, b -> sb.append(String.format("%02x", b)) }
            .toString()
    }
}

